nntrainer/layers/rnn.cpp

```diff
@@ -123,8 +123,8 @@ void RNNLayer::forwarding(bool training) {
       Tensor hs_prev = Tensor(TensorDim(1, 1, 1, oslice.width()),
                               oslice.getAddress(id * oslice.width()));
       // Calculate hs_t = tanh(Whh*h_(t-1) + Wxh*X_t))
-      hs =
-        xs.dot(weight_xh).add(hs_prev.dot(weight_hh).add(bias_h)).apply(tanh);
+      hs = xs.dot(weight_xh).add(hs_prev.dot(weight_hh).add(bias_h));
+      acti_func.run_fn(hs, hs);
```

Comment:
Please add a check in the graph that the next layer of the RNN must not be an activaiton layer.

Reply:
In network graph, there is realizeActivationLayer to create additional activation layer. but for RNN Layer, there is no need to do it.. so.. it just set default tanh if type is ACT_NONE. Talking about the user intended addition activation layer.. I think we don't need to care about. That's user intension. Also to calculate softmax.. user also have responsibility to add Fully Connected Layer. I think the restriction of activation layer is totally depending on user.

Reply:
The issue is that RNN uses activation function internally which is being operated in-place in the current code.
So, if an activation layer is added after it, which is also optimized to work in-place, it will create issues.
If there is no such restriction of activation layer being added to the graph, please add a `TODO:` in `inPlaceOptimize` in the graph to ensure that if the input layer of activation is RNN or RNN has activation internally (to be more precise), then that activation must not be inplace. This will later be updated to each layer provided `isOperatingInPlace` with #867.

Reply:
Let's see. how it works. RNNLayer still in the middle of development. so this kind of optimization can be considered later.

Reply:
Alright. Please add a TODO as comments to keep it as a reminder.